sycl/source/detail/scheduler/commands.cpp

```diff
@@ -504,7 +505,8 @@ Command *Command::processDepEvent(EventImplPtr DepEvent, const DepDesc &Dep) {
 
   // Do not add redundant event dependencies for in-order queues.
   if (Dep.MDepCommand && Dep.MDepCommand->getWorkerQueue() == WorkerQueue &&
-      WorkerQueue->has_property<property::queue::in_order>())
+      WorkerQueue->has_property<property::queue::in_order>() &&
+      getType() != CommandType::HOST_TASK)
```

Comment:
Can we get here if `getType() == CommandType::HOST_TASK`? It looks like that in this case we should exist the function on line 501.

Reply:
Why? I changed that condition as well.